src/NewTools-Finder/StFinderModel.class.st

```diff
@@ -65,9 +65,9 @@ StFinderModel >> application: anApplication [
 StFinderModel >> availablePackages [
 	"Returns for searching available packages by name."
 
-	^ searchEnvironment packages
-		  collect: [ :package | package name ]
-		  as: OrderedCollection
+	| globalEnv |
+	globalEnv := RBBrowserEnvironment new.
+	^ (globalEnv forPackages: globalEnv packages) packages collect: [ :package | package name ] as: OrderedCollection
```

Comment:
This is the main change of the PR. Most of the other changes are to move the behavior of environment selection in StFinderEnvironmentBar instead of StFinderPresenter.

Before we looked in a filtered environment, now we look in an environment having all the packages of the image